File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    
    if n == 2: return 1
    if n == 3: return 2
    
    num1 = 1
    num2 = 2
    
    for i in range(4, n+1):
        current_n = num1 + num2 
        num1 = num2
        num2 = current_n
    
    answer = current_n%1234567
    return answer

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    
    if (n == 2) return 1;
    if (n == 3) return 2;
    
    int num1 = 1;
    int num2 = 2;
    int current_n = 3;
    
    for (int i=4; i<=n; i++){
        current_n = (num1 + num2) % 1234567;
        num1 = num2;
        num2 = current_n;
    }
    
    int answer = current_n % 1234567;
    return answer;
}